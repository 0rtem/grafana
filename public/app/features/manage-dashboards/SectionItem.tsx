import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import FormSwitch from '../../core/components/FormSwitch/FormSwitch';
import { DashboardSectionItem } from '../../types';
import { addTagFilter, setSectionItemSelected } from './state/actions';
import { updateLocation } from '../../core/actions';

export interface Props {
  item: DashboardSectionItem;
  setSectionItemSelected: typeof setSectionItemSelected;
  addTagFilter: typeof addTagFilter;
  updateLocation: typeof updateLocation;
}

export class SectionItem extends PureComponent<Props> {
  toggleSectionItemSelection = event => {
    event.stopPropagation();
    const { item, setSectionItemSelected } = this.props;

    setSectionItemSelected(item);
  };

  selectTag = (tag, event) => {
    event.stopPropagation();
    this.props.addTagFilter(tag);
  };

  navigateToDashboard = () => {
    this.props.updateLocation({ path: this.props.item.url });
  };

  render() {
    const { item } = this.props;

    return (
      <a className={`search-item search-item--indent ${item.checked ? 'selected' : ''}`} href={item.url} target="_self">
        <FormSwitch
          label=""
          onChange={event => this.toggleSectionItemSelection(event)}
          checked={item.checked}
          switchClass="gf-form-switch--transparent gf-form-switch--search-result__item"
        />
        <span className="search-item__icon" onClick={this.navigateToDashboard}>
          <i className="gicon mini gicon-dashboard-list" />
        </span>
        <span className="search-item__body" onClick={this.navigateToDashboard}>
          <div className="search-item__body-title">{item.title}</div>
        </span>
        <span className="search-item__tags">
          {item.tags.map((tag, index) => {
            return (
              <span
                key={index}
                tag-color-from-name="tag"
                className="label label-tag"
                onClick={event => this.selectTag(tag, event)}
              >
                {tag}
              </span>
            );
          })}
        </span>
      </a>
    );
  }
}

const mapDispatchToProps = {
  addTagFilter,
  setSectionItemSelected,
  updateLocation,
};

export default connect(() => {
  return {};
}, mapDispatchToProps)(SectionItem);
