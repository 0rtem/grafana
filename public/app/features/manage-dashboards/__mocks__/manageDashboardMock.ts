import { DashboardSection, DashboardSectionItem } from '../../../types';

export const getMockSections = (amount: number): DashboardSection[] => {
  const sections: DashboardSection[] = [];

  for (let i = 1; i <= amount; i++) {
    sections.push({
      id: i,
      uid: `asdf${i}`,
      title: 'Such a cool folder',
      expanded: false,
      items: [] as DashboardSectionItem[],
      url: `some/url/${i}`,
      icon: 'cool-icon',
      score: i,
      hideHeader: false,
      checked: false,
    });
  }

  return sections;
};

export const getMockSectionsWithItems = (amount: number): DashboardSection[] => {
  return getMockSections(amount).map(section => {
    section.items = getMockSectionItems(amount);

    return section;
  });
};

export const getMockSection = (): DashboardSection => {
  return {
    id: 1,
    uid: 'asdf1',
    title: 'Such a cool folder',
    expanded: false,
    items: [] as DashboardSectionItem[],
    url: 'some/url/1',
    icon: 'cool-icon',
    score: 1,
    hideHeader: false,
    checked: false,
  };
};

export const getMockSectionItems = (amount: number): DashboardSectionItem[] => {
  const items: DashboardSectionItem[] = [];

  for (let i = 0; i <= amount; i++) {
    items.push({
      id: i,
      uid: `id-${i}`,
      title: `item-${i}`,
      uri: '',
      url: `some/url/${i}`,
      type: '',
      tags: ['blue', 'green'],
      isStarred: false,
      folderId: 1,
      folderUid: 'folder-1',
      folderTitle: 'folderur',
      folderUrl: 'some/folderUrl/1',
      checked: false,
    });
  }

  return items;
};
