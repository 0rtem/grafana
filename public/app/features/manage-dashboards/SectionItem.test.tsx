import React from 'react';
import { shallow } from 'enzyme';
import { Props, SectionItem } from './SectionItem';
import { getMockSectionItems } from './__mocks__/manageDashboardMock';

const setup = (propOverrides?: object) => {
  const props: Props = {
    item: getMockSectionItems(1)[0],
    addTagFilter: jest.fn(),
    updateLocation: jest.fn(),
    setSectionItemSelected: jest.fn(),
  };

  Object.assign(props, propOverrides);

  const wrapper = shallow(<SectionItem {...props} />);
  const instance = wrapper.instance() as SectionItem;

  return {
    wrapper,
    instance,
  };
};

describe('Render', () => {
  it('should render component', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
