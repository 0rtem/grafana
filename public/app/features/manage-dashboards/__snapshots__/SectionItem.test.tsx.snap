// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<a
  className="search-item search-item--indent "
  href="some/url/0"
  target="_self"
>
  <FormSwitch
    checked={false}
    label=""
    onChange={[Function]}
    switchClass="gf-form-switch--transparent gf-form-switch--search-result__item"
  />
  <span
    className="search-item__icon"
    onClick={[Function]}
  >
    <i
      className="gicon mini gicon-dashboard-list"
    />
  </span>
  <span
    className="search-item__body"
    onClick={[Function]}
  >
    <div
      className="search-item__body-title"
    >
      item-0
    </div>
  </span>
  <span
    className="search-item__tags"
  >
    <span
      className="label label-tag"
      key="0"
      onClick={[Function]}
      tag-color-from-name="tag"
    >
      blue
    </span>
    <span
      className="label label-tag"
      key="1"
      onClick={[Function]}
      tag-color-from-name="tag"
    >
      green
    </span>
  </span>
</a>
`;
